cpp/src/tests/centrality/katz_centrality_test.cu

```diff
@@ -0,0 +1,131 @@
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+#include "gmock/gmock-generated-matchers.h"
+#include "high_res_clock.h"
+#include "cuda_profiler_api.h"
+#include <cugraph.h>
+#include "test_utils.h"
+#include <thrust/device_ptr.h>
+#include <fstream>
+
+std::vector<int>
+getGoldenTopKIds(std::string file, int k = 10) {
+  std::vector<int> vec;
+  std::ifstream fin(file);
+  int val;
+  while (fin>>val) {
+    vec.push_back(val);
+  }
+  vec.resize(k);
+  return vec;
+}
+
+std::vector<int>
+getTopKIds(gdf_column_ptr katz, int k = 10) {
+  int count = katz.get()->size;
+  cudaStream_t stream = nullptr;
+  rmm::device_vector<int> id(count);
+  thrust::sequence(rmm::exec_policy(stream)->on(stream), id.begin(), id.end());
+  auto colptr = thrust::device_pointer_cast(reinterpret_cast<double*>(katz.get()->data));
+  thrust::sort_by_key(rmm::exec_policy(stream)->on(stream),
+      colptr, colptr + count, id.begin(), thrust::greater<double>());
+  std::vector<int> topK(k);
+  thrust::copy(id.begin(), id.begin() + k, topK.begin());
+  return topK;
+}
+
+typedef struct CoreNumber_Usecase_t {
+  std::string matrix_file;
+  std::string result_file;
+  CoreNumber_Usecase_t(const std::string& a, const std::string& b) {
+    // assume relative paths are relative to RAPIDS_DATASET_ROOT_DIR
+    const std::string& rapidsDatasetRootDir = get_rapids_dataset_root_dir();
+    if ((a != "") && (a[0] != '/')) {
+      matrix_file = rapidsDatasetRootDir + "/" + a;
+    } else {
+      matrix_file = a;
+    }
+    if ((b != "") && (b[0] != '/')) {
+      result_file = rapidsDatasetRootDir + "/" + b;
+    } else {
+      result_file = b;
+    }
+  }
+  CoreNumber_Usecase_t& operator=(const CoreNumber_Usecase_t& rhs) {
+    matrix_file = rhs.matrix_file;
+    result_file = rhs.result_file;
+    return *this;
+  }
+} CoreNumber_Usecase;
+
+class Tests_CoreNumber : public ::testing::TestWithParam<CoreNumber_Usecase> {
```

Comment:
Just for curiosity, why did you name this class as Tests_"CoreNumber"?

Reply:
Fixed